src/js/tools/superselect.js

```diff
@@ -76,23 +76,21 @@ define(function (require, exports, module) {
         } else {
             var currentLayers = currentDocument.layers.selected,
                 currentLayer = currentLayers.first();
-            
-            if (currentLayer.vectorMaskEnabled && !currentLayer.vectorMaskEmpty) {
-                return descriptor.playObject(vectorMaskLib.activateVectorMaskEditing())
+
+            if (currentLayer.vectorMaskEnabled) {
+                this.flux.actions.layers.resetLayers(currentDocument, currentLayer, false, false, true)
```

Comment:
I think this should be a transfer effectively transferring from `tools.select`? Or is there some reason that this just enqueues an action instead of transferring? Do you want `tools.select` to finish before this finishes? (I realize that you explicitly do want that for the result of `enterPathModalState` below, but it's not clear if similar reasoning applies here.)